esrally/reporter.py

```diff
@@ -60,15 +60,15 @@ def summarize(race, cfg, lap=None):
 
 
 def compare(cfg):
-    baseline_ts = cfg.opts("reporting", "baseline.timestamp")
-    contender_ts = cfg.opts("reporting", "contender.timestamp")
+    baseline_id = cfg.opts("reporting", "baseline.id")
+    contender_id = cfg.opts("reporting", "contender.id")
 
-    if not baseline_ts or not contender_ts:
+    if not baseline_id or not contender_id:
         raise exceptions.SystemSetupError("compare needs baseline and a contender")
     race_store = metrics.race_store(cfg)
     ComparisonReporter(cfg).report(
```

Comment:
I think in the output we should also show the `Race ID` (in addition to `Race timestamp`). Here is an example output (of the current state on this branch):

```
Comparing baseline
  Race timestamp: 2019-07-16 07:59:59
  Challenge: append-no-conflicts-index-only
  Car: defaults+basic-license

with contender
  Race timestamp: 2019-07-16 08:01:16
  Challenge: append-no-conflicts-index-only
  Car: defaults+basic-license
```

IMHO we should add in the first line also the Race ID` (as this is what the user has provided on the command line)